include/cuco/detail/probing_scheme_impl.inl

```diff
@@ -100,7 +100,7 @@ __host__ __device__ constexpr auto linear_probing<CGSize, Hash>::operator()(
 {
   using size_type = typename Extent::value_type;
   return detail::probing_iterator<Extent>{
-    cuco::detail::sanitize_hash<size_type>(hash_(probe_key)) % upper_bound,
+    cuco::detail::sanitize_hash<size_type>(hash_(probe_key)) % static_cast<size_type>(upper_bound),
```

Comment:
IIUC the `static_cast` demotes the `fast_int` to a simple integer in the dynamic extent case. This eliminates the fast modulo optimization.

Reply:
Yeah, that's a bit inconvenient since a static window extent is no longer implicitly convertible to a plain int but at least we get uniform behavior that way. I'm open to suggestions how to improve this.